src/js/samples/components/Sidebar/SampleSidebarSelector.tsx

```diff
@@ -0,0 +1,95 @@
+import { fontWeight, getFontSize } from "@app/theme";
+import { BoxGroupSearch, Icon, SidebarHeaderButton } from "@base";
+import { useFuse } from "@base/hooks";
+import { Popover } from "@base/Popover";
+import { Label } from "@labels/types";
+import { SubtractionShortlist } from "@subtraction/types";
+import React from "react";
+import { Link } from "react-router-dom";
+import styled from "styled-components";
+import { SampleSidebarSelectorItem } from "./SampleSidebarSelectorItem";
+
+const SampleSidebarSelectorButton = styled.div`
+    display: flex;
+    border-top: 1px solid;
+    border-color: ${props => props.theme.color.greyLight};
+    width: 100%;
+    align-items: flex-end;
+
+    a {
+        margin-left: auto;
+        font-size: ${getFontSize("md")};
+        font-weight: ${fontWeight.thick};
+        padding: 10px 10px 10px 0px;
+    }
+`;
+
+const SampleItemComponentsContainer = styled.div`
+    max-height: 300px;
+    overflow-y: scroll;
+`;
+
+type SampleSidebarSelectorProps = {
+    /** The link to manage labels or subtractions */
+    manageLink: string;
+    /** A callback function to handle sidebar item selection */
+    onUpdate: (id: string | number) => void;
+    /** List of label ids applied to some, but not all selected samples */
+    partiallySelectedItems?: number[];
+    /** The styled component for the list items */
+    render: (result: { color?: string; description?: string; name: string }) => React.ReactNode;
+    /** A list of labels or default subtractions */
+    sampleItems: Label[] | SubtractionShortlist[];
+    /** A list of selected items by their ids */
+    selectedItems: string[] | number[];
+    /** Whether the sidebar is labels or subtractions */
+    selectionType: string;
+};
+
+/**
+ * Displays a dropdown list of labels or subtractions
+ */
+export function SampleSidebarSelector({
+    render,
+    sampleItems,
+    selectedItems,
+    partiallySelectedItems = [],
+    onUpdate,
+    selectionType,
+    manageLink,
+}: SampleSidebarSelectorProps) {
+    const [results, term, setTerm] = useFuse(sampleItems, ["name"], []);
+    const sampleItemComponents = results.map((item: Label | SubtractionShortlist) => (
+        <SampleSidebarSelectorItem
+            key={item.id}
+            selected={
+                selectionType === "labels"
+                    ? (selectedItems as number[]).includes(item.id as number)
+                    : (selectedItems as string[]).includes(item.id as string)
```

Comment:
I would give this a rework using generics. You should be able to declare a type such that is either a string or number but only one of the two. As written this introduces an`if` into the runtime,  while the types are removed at compile time.